microsoft/testsuites/performance/networkperf.py

```diff
@@ -0,0 +1,216 @@
+# Copyright (c) Microsoft Corporation.
+# Licensed under the MIT license.
+from typing import Any, Dict, List, cast
+
+from lisa import (
+    Logger,
+    RemoteNode,
+    TestCaseMetadata,
+    TestSuite,
+    TestSuiteMetadata,
+    notifier,
+    simple_requirement,
+)
+from lisa.environment import Environment
+from lisa.features import Sriov, Synthetic
+from lisa.schema import NetworkDataPath
+from lisa.tools import Lagscope, Lscpu, Netperf, Sar, Ssh, Sysctl
+from lisa.util import dict_to_fields
+from microsoft.testsuites.network.common import stop_firewall
+from microsoft.testsuites.performance.common import cleanup_process
+
+
+@TestSuiteMetadata(
+    area="network",
+    category="performance",
+    description="""
+    This test suite is to validate linux network performance.
+    """,
+)
+class NetworkPerformace(TestSuite):
+    @TestCaseMetadata(
+        description="""
+        This test case uses lagscope to test synthetic network latency.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            min_core_count=72,
+            network_interface=Synthetic(),
+        ),
+    )
+    def perf_tcp_latency_synthetic(self, environment: Environment) -> None:
+        self.perf_tcp_latency(environment, "perf_tcp_latency_synthetic")
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses lagscope to test sriov network latency.
+        """,
+        priority=2,
+        requirement=simple_requirement(
+            min_count=2,
+            min_core_count=72,
+            network_interface=Sriov(),
+        ),
+    )
+    def perf_tcp_latency_sriov(self, environment: Environment) -> None:
+        self.perf_tcp_latency(environment, "perf_tcp_latency_sriov")
+
+    @TestCaseMetadata(
+        description="""
+        This test case uses sar to test synthetic network PPS (Packets Per Second)
+         when running netperf with single port.
+        """,
+        priority=3,
+        requirement=simple_requirement(
+            min_count=2,
+            min_core_count=32,
```

Comment:
Can we have the same criteria for them? some are 72 cores, some are 36 cores. The result is not comparable between sar and other tools.

Reply:
I followed the same setting with lisav2.

Reply:
As another comment about the core count, let's remove this limitation in test case level. We can specify the vm size in pipeline.

Reply:
remove core count